### DISCUSSION TITLE
What's the "default currency pairs" setting for?

### DISCUSSION THREAD

**Discussion by ksdhans** [btcpayserver/btcpayserver#3226] [Technical Support]
What's the "default currency pairs" setting for? Is it needed when all invoices come from WooCommerce?

I haven't been able to find any documentation on it, or the "Add a spread on exchange rate of ... %" setting either.

One concern I have, is that I some of the currency pairs I want don't work. For example, LTC_USD doesn't work, and XMR_USD doesn't work (or XMR_anything). If BTCPay can't get the exchange rate from LTC/XMR to fiat currencies, then I don't see how those cryptocurrencies can be used.

**Comment by bolatovumar** [comment]
@ksdhans "Default currency pairs" setting seems to be for when you use the `api/rates` endpoint without explicitly specifying the currency pairs in the request body. If you specify default currency pairs then they will be used.

"Add a spread on exchange rate of ... %" setting will decrease the bid price by specified % and increase the ask price by % percent effectively making the price at which you sell BTC (or anything else) higher and price at which you are willing to buy BTC lower. See this comment for a good explanation: https://github.com/btcpayserver/btcpayserver/issues/2658#issuecomment-876869351.

> For example, LTC_USD doesn't work, and XMR_USD doesn't work

You probably need to pick a different price source for these. It works on my end.

![Capture](https://user-images.githubusercontent.com/1934678/149269558-79322704-b487-4764-ae4b-666d8433648d.PNG)

**Reply by ksdhans** [reply]
Thanks for the explanation. 

I tried OKCoin, and indeed it works with LTC_USD. However, XMR was missing. That's only available on Kraken & Poloniex (and maybe one or two others. Unfortunately, they don't support other fiat currencies. However, I realized that this is what the "advanced rate rules" scripting feature is for. 

If anyone else is looking for XMR to an other currency, here's an  "advanced rate rules" script that'll get it working:

// Default rate rules for LTC
LTC_X = LTC_BTC * BTC_X;
LTC_BTC = coingecko(LTC_BTC);
// Default rate rules for XMR
XMR_X = XMR_BTC * BTC_X;
XMR_BTC = kraken(XMR_BTC);
////////
X_X = coingecko(X_X);

With that script, you can also get XMR_X (where X is anything else).
